src/oracle/OracleWriter.sol

```diff
@@ -1,16 +1,56 @@
 pragma solidity 0.8.13;
 
-import "src/interfaces/IOracleWriter.sol";
+import { IOracleWriter } from "src/interfaces/IOracleWriter.sol";
+import { LogCompression } from "src/libraries/LogCompression.sol";
+import { stdMath } from "forge-std/Test.sol";
 
 abstract contract OracleWriter is IOracleWriter {
+    // 10 basis points per second which is 6% per minute and doubling of the price in 16 minutes
+    uint256 internal constant MAX_CHANGE_PER_SEC = 0.001e18;
+
     Observation[65536] public observations;
     uint16 public index = type(uint16).max;
 
+    // maximum allowed rate of change of price per second
+    // for clamping the price
+    uint64 internal allowedChangePerSecond = 0.0005e18; // 5 bp
+    uint256 public prevClampedPrice;
```

Comment:
Seems like we need to write a whole extra word for every swap? How many more `SSTORE`s is the new branch, also what is the gas difference?

Reply:
as discussed, I've made a perf note to try read the two previous observations to figure out the clamped price instead of using this new variable